File: sdks/go/pkg/models/shared/comment.go
package shared

import (
	"time"
)

// CommentReference
// The bare basic reference for an object
type CommentReference struct {
	ID   *string `json:"id,omitempty"`
	Type *string `json:"type,omitempty"`
}

type CommentTypeEnum string

const (
	CommentTypeEnumComment CommentTypeEnum = "comment"
)

// Comment
// Base representation of a comment.
type Comment struct {
	CreatedAt      *time.Time        `json:"created_at,omitempty"`
	CreatedBy      *UserMini         `json:"created_by,omitempty"`
	ID             *string           `json:"id,omitempty"`
	IsReplyComment *bool             `json:"is_reply_comment,omitempty"`
	Item           *CommentReference `json:"item,omitempty"`
	Message        *string           `json:"message,omitempty"`
	ModifiedAt     *time.Time        `json:"modified_at,omitempty"`
	Type           *CommentTypeEnum  `json:"type,omitempty"`
}

File: sdks/java/lib/src/main/java/models/shared/Comment.java
package .models.shared;

import java.time.OffsetDateTime;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import .utils.DateTimeSerializer;
import .utils.DateTimeDeserializer;

/**
 * Comment
 * Base representation of a comment.
**/
public class Comment {
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("created_at")
    public OffsetDateTime createdAt;
    public Comment withCreatedAt(OffsetDateTime createdAt) {
        this.createdAt = createdAt;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("created_by")
    public UserMini createdBy;
    public Comment withCreatedBy(UserMini createdBy) {
        this.createdBy = createdBy;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("id")
    public String id;
    public Comment withId(String id) {
        this.id = id;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("is_reply_comment")
    public Boolean isReplyComment;
    public Comment withIsReplyComment(Boolean isReplyComment) {
        this.isReplyComment = isReplyComment;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("item")
    public CommentReference item;
    public Comment withItem(CommentReference item) {
        this.item = item;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("message")
    public String message;
    public Comment withMessage(String message) {
        this.message = message;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("modified_at")
    public OffsetDateTime modifiedAt;
    public Comment withModifiedAt(OffsetDateTime modifiedAt) {
        this.modifiedAt = modifiedAt;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("type")
    public CommentTypeEnum type;
    public Comment withType(CommentTypeEnum type) {
        this.type = type;
        return this;
    }
}